### PR TITLE
[SOAR-9920] AbuseIPDB Connect Test Error Handling

### PR DESCRIPTION
**Pull request by ablakley-r7**
### Description

Describe the proposed changes:

  - Failed connection tests are returning a 200 status code in Insight Connect. This should be updated to return the proper status code corresponding to a failure.

## PR Requirements

Developers, verify you have completed the following items by checking them off:

#### Unit Tests

Review our documentation on [generating](https://docs.rapid7.com/insightconnect/unit-test-generation) and [writing](https://docs.rapid7.com/insightconnect/unit-test-primer) plugin unit tests

- [ ] Unit tests written for any new or updated code

#### In-Product Tests

If you are an InsightConnect customer or have access to an InsightConnect instance, the following in-product tests should be done:

- [ ] Screenshot of job output with the plugin changes
- [ ] Screenshot of the changed connection, actions, or triggers input within the InsightConnect workflow builder

### Style

Review the [style guide](https://docs.rapid7.com/insightconnect/style-guide/)

- [ ] For dependencies, pin [OS package](https://docs.rapid7.com/insightconnect/style-guide/#dockerfile) and [Python package](https://docs.rapid7.com/insightconnect/style-guide/#requirements.txt) versions
- [ ] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [ ] For size, use the [slim SDK images](https://docs.rapid7.com/insightconnect/sdk-guide/#sdk-guide) when possible: ``komand/python-3-37-slim-plugin`` and ``komand/python-3-37-plugin``
- [ ] For error handling, use of [PluginException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations/#plugin-exceptions) and [ConnectionTestException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations#connection-exceptions)
- [ ] For logging, use [self.logger](https://docs.rapid7.com/insightconnect/sdk-guide/#logging)
- [ ] For docs, use [changelog style](https://docs.rapid7.com/insightconnect/style-guide/#changelog)
- [ ] For docs, validate markdown with ``make validate`` which calls ``mdl`` to lint ``help.md``

### Functional Checklist
- [ ] Work fully completed
- [ ] Functional
  - [ ] Any new actions/triggers include JSON [test files](https://docs.rapid7.com/insightconnect/style-guide/#tests) in the `tests/` directory created with `icon-plugin run -c sample $action > tests/$action.json`
  - [ ] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [ ] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -T all --debug --jq` (use PR format at end)
  - [ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -R all --debug --jq` (use PR format at end)

### Assessment

You must validate your work to reviewers:

1. Run `make validate` and make sure everything passes
2. Run the assessment tool: `icon-plugin run -A -R all -T all`. For single action validation: `icon-plugin run -A -R tests/my_action.json -T tests/my_action.json`
3. Copy (`icon-plugin ... | pbcopy`) and paste the output in **a new post** on this PR
4. Add required screenshots from the In-Product Tests section

### CODE REVIEW

File: plugins/abuseipdb/komand_abuseipdb/connection/connection.py [hunk 1]
```diff
@@ -8,32 +8,36 @@
...
-            raise
-
-        return json_
+  def test(self):
+    # Use private IP Addresses for testing the API (e.g. 127.0.0.1) from https://www.abuseipdb.com/api
+    url = "https://api.abuseipdb.com/api/v2/check"
+    params = {"ipAddress": "127.0.0.0"}
+    try:
+      r = requests.get(url, params=params, headers=self.headers)
+      json_ = r.json()
```
Comment (igorski-r7) [inline plugins/abuseipdb/komand_abuseipdb/connection/connection.py:29]: It's up to you 😄
```suggestion
      response = requests.get(url, params=params, headers=self.headers)
      json_ = response.json()
```

File: plugins/abuseipdb/komand_abuseipdb/connection/connection.py [hunk 2]
```diff
@@ -8,32 +8,36 @@
...
+    except json.decoder.JSONDecodeError:
+      raise ConnectionTestException(
+        cause="Received an unexpected response from AbuseIPDB.",
+        assistance=f"(non-JSON or no response" f" was received). Response was: {r.text}",
```
Comment (igorski-r7) [inline plugins/abuseipdb/komand_abuseipdb/connection/connection.py:33]:
```suggestion
        assistance=f"(non-JSON or no response" f" was received). Response was: {response.text}",
```

File: plugins/abuseipdb/komand_abuseipdb/connection/connection.py [hunk 3]
```diff
@@ -8,32 +8,36 @@
...
+      )
+    except Exception as e:
+      self.logger.error(e)
```
Comment (igorski-r7) [inline plugins/abuseipdb/komand_abuseipdb/connection/connection.py:36]:
```suggestion
    except Exception as error:
      self.logger.error(error)
```

File: plugins/abuseipdb/komand_abuseipdb/connection/connection.py [hunk 4]
```diff
@@ -8,32 +8,36 @@
...
+      raise
+    if r.status_code not in range(200, 299):
```
Comment (igorski-r7) [inline plugins/abuseipdb/komand_abuseipdb/connection/connection.py:38]:
```suggestion
    if response.status_code not in range(200, 299):
```

File: plugins/abuseipdb/komand_abuseipdb/connection/connection.py [hunk 5]
```diff
@@ -8,32 +8,36 @@
...
+      raise ConnectionTestException(
+        cause="Received an unexpected response from AbuseIPDB.",
+        assistance=f"(unexpected response was received). Response was: {r.text}; Status Code was: {r.status_code}",
```
Comment (igorski-r7) [inline plugins/abuseipdb/komand_abuseipdb/connection/connection.py:41]:
```suggestion
        assistance=f"(unexpected response was received). Response was: {response.text}; Status Code was: {response.status_code}",
```

**Review by igorski-r7 — APPROVED** [review]
Looks great! Left some comments, but it's up to you for changing this 😄

**Review by cmcnally-r7 — APPROVED** [review]
LGTM 👍

**Review by igorski-r7 — APPROVED** [review]

**Pull request merged**